src/lib.rs

```diff
@@ -74,7 +77,7 @@ pub trait Database: plumbing::DatabaseOps {
     /// This function is invoked when a dependent query is being computed by the
     /// other thread, and that thread panics.
     fn on_propagated_panic(&self) -> ! {
-        panic!("concurrent salsa query panicked")
+        Canceled::throw()
     }
```

Comment:
Seems like we should remove this callback from the public API altogether? It was added specifically for cancellation.